tests/unit/query_generators/test_condensed_query_generator.py

```diff
@@ -0,0 +1,85 @@
+from typing import List
+from unittest.mock import create_autospec
+
+import pytest
+
+from canopy.chat_engine.query_generator import CondensedQueryGenerator
+from canopy.llm import BaseLLM
+from canopy.models.api_models import ChatResponse, _Choice, TokenCounts
+from canopy.models.data_models import Query, UserMessage, AssistantMessage
+
+
+class TestCondensedQueryGenerator:
+
+    @staticmethod
+    @pytest.fixture
+    def mock_llm():
+        return create_autospec(BaseLLM)
+
+    @staticmethod
+    @pytest.fixture
+    def query_generator(mock_llm):
+        query_gen = CondensedQueryGenerator(
+            llm=mock_llm,
+        )
+        return query_gen
+
+    @staticmethod
+    @pytest.fixture
+    def sample_messages():
+        return [UserMessage(content="How can I init a client?"),
+                AssistantMessage(content="Which kind of client?"),
+                UserMessage(content="A pinecone client.")]
+
+    @staticmethod
+    @pytest.mark.parametrize(("response", "query"), [
+        (
+                '{"question": "How do I init a pinecone client?"}',
+                "How do I init a pinecone client?"),
+        (
+                '...',
```

Comment:
I'm not sure I'm getting this expected result (`'...'`), can you please explain?

Reply:
I wanted to simulate that if the model return a non-parsable json we receive the last message as an answer. I will change it to something more understandable.